modules/collator-selection/src/lib.rs

```diff
@@ -404,26 +404,28 @@ pub mod pallet {
 		}
 
 		fn start_session(index: SessionIndex) {
-			let (candidates_len, collator_len) = <Candidates<T>>::mutate(|candidates| -> (usize, usize) {
-				let mut collator = 0;
-				candidates.iter().for_each(|candidate| {
-					if index >= candidate.start_session {
-						collator += 1;
-						<SessionPoints<T>>::insert(&candidate.who, 0);
-					}
-				});
-				(candidates.len(), collator)
+			let validators = T::ValidatorSet::validators();
+			let candidates = Self::candidates();
+			let mut collators = vec![];
+
+			candidates.iter().for_each(|candidate| {
+				let who = ValidatorIdOf::<T>::convert(candidate.who.clone())
```

Comment:
I think you can avoid this by having `type ValidatorSet: ValidatorSet<Self::AccountId, ValidatorId = Self::AccountId>;`